src/charm.py

```diff
@@ -159,173 +139,43 @@ def _check_status(self) -> bool:
-        ca_cert = None
-        if self.etcd_use_tls:
-            protocol = "https"
-            ca_cert = Path("/etc/slurm/tls_cert.crt")
-            ca_cert.write_text(self.etcd_tls_cert)
-            ca_cert = Path.as_posix()
-        if self.etcd_ca_cert:
-            ca_cert = Path("/etc/slurm/ca_cert.crt")
-            ca_cert.write_text(self.etcd_ca_cert)
-            ca_cert = Path.as_posix()
-
-        logger.debug(f"## Connecting to etcd3 in {protocol}://{host}:{port}, {ca_cert}")
-        client = Etcd3AuthClient(
-            host=host,
-            port=port,
-            protocol=protocol,
-            ca_cert=ca_cert,
-            username=username,
-            password=password,
-        )
-
-        logger.debug("## Querying etcd3 for node list")
-        try:
-            v = client.get(key="nodes/all_nodes")
-            logger.debug(f"## Got: {v}")
-        except Exception as e:
-            logger.error(f"## Unable to connect to {host} to get list of nodes: {e}")
-            event.defer()
-            return
-
-        node_accounted = False
-        if v:
-            hostnames = json.loads(v[0])
-            logger.debug(f"### etcd3 node list: {hostnames}")
-            if self.hostname in hostnames:
-                self.on.slurmctld_started.emit()
-                node_accounted = True
-
-        if not node_accounted:
-            logger.debug("## Node not accounted for. Deferring.")
-            event.defer()
-
     def _on_slurmctld_unavailable(self, event):
         logger.debug("## Slurmctld unavailable")
         self._set_slurmctld_available(False)
-        self._set_slurmctld_started(False)
-        self._slurm_manager.slurm_systemctl("stop")
+        slurmd.stop()
         self._check_status()
 
-    def _on_slurmctld_started(self, event):
-        """Set flag to True and emit slurmd_start event."""
-        self._set_slurmctld_started(True)
-        self.on.slurmd_start.emit()
-
-    def _on_slurmd_start(self, event):
-        if not self._check_status():
-            event.defer()
-            return
-
-        # only set up fluentbit if we have a relation to it
-        if self._fluentbit._relation is not None:
-            self._configure_fluentbit()
+    def _on_slurmd_started(self, _: ServiceStartedEvent) -> None:
+        """Handle event emitted by systemd after slurmd daemon successfully starts."""
+        self.unit.status = ActiveStatus()
 
-        # at this point, we have slurm installed, munge configured, and we know
-        # slurmctld accounted for this node. It should be safe to start slurmd
-        if self.ensure_slurmd_starts():
-            logger.debug("## slurmctld started and slurmd is running")
-        else:
-            event.defer()
-        self._check_status()
+    def _on_slurmd_stopped(self, _: ServiceStoppedEvent) -> None:
+        """Handle event emitted by systemd after slurmd daemon is stopped."""
+        self.unit.status = BlockedStatus("slurmd off")
```

Comment:
Sure - there may be less ambiguity to say that slurmd is not running since some folks may think off means that the node is just not communicating with the slurmctld service.

Reply:
Updated in commit https://github.com/omnivector-solutions/slurmd-operator/pull/21/commits/16d137f544bf99c2fdfdfa462eef68c216e17769